arroyo-api/src/rest_types.rs

```diff
@@ -31,6 +41,68 @@ pub struct Pipeline {
     pub checkpoint_interval_micros: u64,
     pub stop: StopType,
     pub created_at: u64,
+    pub action: Option<StopType>,
```

Comment:
The `action` fields are pretty specific to powering the UI of the console and exist to shift some complex logic from the console to the server. I think it would be good to avoid documenting them as part of the public API as they are basically derivative of other information and could easily change in the future if the UI changes.

I know it's a bit impure but is there a way we could hide these away somewhere?

Reply:
The way we're generating the api spec I don't think allows excluding a field from the spec, which wouldn't really work anyway with openapi-fetch in the frontend. 

Right now the only way I can think of simplifying it is by moving some of the logic to the frontend.